pkg/generators/forwarding/fluentd/syslog_conf.go

```diff
@@ -1,14 +1,199 @@
 package fluentd
 
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// The Syslog output fields can be set to an expression of the form $.abc.xyz
+// If an expression is used, its value will be taken from corresponding key in the record
+var keyre = regexp.MustCompile(`^\$(\.[[:word:]]*)+$`)
+
+var tagre = regexp.MustCompile(`\${tag\[-??\d+\]}`)
+
 func (conf *outputLabelConf) SyslogPlugin() string {
 	if protocol := conf.Protocol(); protocol == "udp" {
 		return "syslog"
 	}
 	return "syslog_buffered"
 }
 
+func (conf *outputLabelConf) Facility() string {
+	facility := strings.ToLower(conf.Target.Syslog.Facility)
+	if facility == "" {
+		return "user"
+	}
+	if conf.isFacilityKeyExpr() {
+		return fmt.Sprintf("${%s}", facility)
+	}
+	return conf.Target.Syslog.Facility
+}
+
+func (conf *outputLabelConf) Severity() string {
+	severity := strings.ToLower(conf.Target.Syslog.Severity)
+	if severity == "" {
+		return "debug"
```

Comment:
Is this default a general default for syslog or a tradeoff for our API?

Reply:
this is for our API. there is no default in syslog spec.